File: November/Day6/orderly_queue.py
# https://leetcode.com/problems/orderly-queue/description/

class Solution:
    def orderlyQueue(self, s: str, k: int) -> str:
        if k > 1:
            return "".join(sorted(s))
        
        else:
            ans = s
            for _ in range(len(s)):
                s = s[1:] + s[0]
                ans = min(ans, s)
        
        return ans

File: November/Day6/orderlyQueue.java
class Solution {
    public String orderlyQueue(String s, int k) {
        if (k > 1) {
            char arr[] = s.toCharArray();
            Arrays.sort(arr);
            return new String(arr);
        }
        String ans = s;
        for (int i = 1; i < s.length(); i++) {
            String tmp = s.substring(i) + s.substring(0, i);
            if (tmp.compareTo(ans) < 0) ans = tmp;
        }
        return ans;
    }
}